File: employeef/src/app/employee.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { map, Observable } from 'rxjs';
import { Employee, StandResponse } from './employee/employee.modal';

@Injectable({
  providedIn: 'root'
})
export class EmployeeService {

  constructor(private http: HttpClient) { }

  apiUrl = "http://localhost:8080/api/v1/employee";

  public saveEmployee(employee: Employee): Observable<Employee> {
    return this.http.post<Employee>(`${this.apiUrl}/save`, employee);
  }

  public getEmployees(): Observable<Employee[]> {
   
    return this.http.get<StandResponse<Employee[]>>(`${this.apiUrl}/employees`).pipe(
      map(response => response.data) 
    );
  }

  public deleteEmployee(employeeId: number): Observable<string> {
    return this.http.delete<string>(`${this.apiUrl}/delete/${employeeId}`);

  }

  public updateEmployee(employee: Employee): Observable<Employee> {
    return this.http.put<StandResponse<Employee>>(`${this.apiUrl}/update`, employee).pipe(
      map(response => response.data)
    );

  }

  public getEmployee(id : number): Observable<Employee> {
   
    return this.http.get<StandResponse<Employee>>(`${this.apiUrl}/employee/${id}`).pipe(
      map(response => response.data) 
    );
  }
}

File: Backend/employee/employee/src/main/java/com/employee/employee/service/EmployeeService.java
package com.employee.employee.service;

import com.employee.employee.dto.paginated.PaginatedEmployees;
import com.employee.employee.dto.request.EmpSaveReqDto;
import com.employee.employee.dto.request.EmpUpdateReqDto;
import com.employee.employee.dto.response.EmpResponseDto;

import java.util.List;

public interface EmployeeService {
    EmpResponseDto saveEmployee(EmpSaveReqDto empSaveReqDto);

    List<EmpResponseDto> getEmployees();

    EmpResponseDto getEmployee(long employeeId);

    String deleteEmployee(long employeeId);

    EmpResponseDto updateEmployee(EmpUpdateReqDto empUpdateReqDto);

    PaginatedEmployees getEmployeesPage(int page, int size);
}
